File: Server/Dtos/FoodDto.cs
using Server.Models;
using Server.Shared;
using System.Text.Json.Serialization;

namespace Server.Dtos;

// Transport DTO for creating/updating Food (used also in edit requests).
public class FoodDto
{
  public int Id { get; set; }
  public LanguageText Name { get; set; } = new();
  public LanguageText Keys { get; set; } = new();
  public LanguageText Description { get; set; } = new();
  public List<string> Imgs { get; set; } = new();
  [JsonConverter(typeof(JsonStringEnumConverter))]
  public MeasurementUnit MeasurementUnit { get; set; }
  public Measures Measures { get; set; } = new();
  public List<string> Categories { get; set; } = new();
  public int? IconId { get; set; }
  [JsonConverter(typeof(JsonStringEnumConverter))]
  public FoodType Type { get; set; } = FoodType.Solid;
  public NutritionalInformation NutritionalInformation { get; set; } = new();
  public Minerals Minerals { get; set; } = new();
  public Vitamins Vitamins { get; set; } = new();
  public AminoAcids AminoAcids { get; set; } = new();
}

File: common/src/services/food/food.dto.ts
import type { LanguageText } from '../language/language.types';
import type { MeasureType } from '../measure/measure.types';
import type { Measures } from '../measure/measures.types';
import type { NutritionalInformationType } from '../nutritional-information/nutritional-information.types';
import type { MineralType } from '../mineral/mineral.types';
import type { VitaminType } from '../vitamin/vitamin.types';
import type { AminoAcidType } from '../amino-acid/amino-acid.types';
import type { MeasurementUnitDto } from '../measurement-unit/measurement-unit.types';
import { FoodType } from './food.types';

// Nutrient maps keyed by enum string as per API contract
export type NutritionalInformationDto = Partial<
  Record<Lowercase<NutritionalInformationType>, number>
>;

export type MineralsDto = Partial<Record<Lowercase<MineralType>, number>>;

export type VitaminsDto = Partial<Record<Lowercase<VitaminType>, number>>;

export type AminoAcidsDto = Partial<Record<Lowercase<AminoAcidType>, number>>;

export interface FoodDto {
  id?: number;
  name: LanguageText;
  keys: LanguageText;
  description: LanguageText;
  imgs: string[];
  measurementUnit: MeasurementUnitDto;
  measures: Measures;
  categories: string[];
  iconId?: number;
  type: FoodType;
  nutritionalInformation?: NutritionalInformationDto;
  minerals?: MineralsDto;
  vitamins?: VitaminsDto;
  aminoAcids?: AminoAcidsDto;
}
